### ISSUE TITLE
Colorless config

### ISSUE THREAD

**Issue by ema79** [worron/awesome-config#20]
Hello,
while running the colorless config (ubuntu 18.04 - awesome 4 - lua 5.3)  got the following errors:
```
~/.config/awesome/redflat/gauge/separator.lua 	line 	73
~/.config/awesome/redflat/float/tooltip.lua	line 	57
~/.config/awesome/colorless/env-config.lua	line 	90
~/.config/awesome/redflat/util/key.lua 		line 	19
~/.config/awesome/redflat/util/key.lua  	        line 	21
~/.config/awesome/redflat/float/hotkeys.lua	line    99
~/.config/awesome/redflat/float/hotkeys.lua	line   100
~/.config/awesome/redflat/widget/tasklist.lua 	line   259
~/.config/awesome/redflat/widget/tasklist.lua   line   334
```
as in lua 5.2 "unpack" is "table.unpack" I just edited the files and it works.

the next "error report" is the following: 
`~/.config/awesome/redflat/widget/tasklist.lua:186: attempt to index a nil value (field '?' )`

this is the code:
```

-- set layout icon for every tag
local l = awful.layout.getname(awful.tag.getproperty(t, "layout"))

local check_icon = style.micon.blank
if c then
local client_tags = c:tags()
check_icon = awful.util.table.hasitem(client_tags, t) and style.micon.check or check_icon
end

for _, index in ipairs(submenu_index) do
local submenu = menu.items[index].child
if submenu.items[k].icon then
submenu.items[k].icon:set_image(style.layout_icon[l] or style.layout_icon.unknown)
end

```
this is line 186
`local submenu = menu.items[index].child`

can someone help?

tnx

**Comment by worron** [comment]
Just tried ubuntu 18.04 VM and I can't reproduce the last problem. To be sure you didn't some mistake while editing I can recommend make a fresh install of configs. Then add this as first line of config (`rc.lua` or `rc-colorless.lua`)
```lua
unpack = unpack or table.unpack
```
This will resolve `unpack` issue without any additional editing. Then report if `tasklist` error  still appears.

**Comment by ema79** [comment]
yeah, tnx ;)
I just put the `unpack hack` in every rc.theme.lua and now the colorless theme seems to be fully working.

I just did a test on other confs too, but there I stil got some error reports. The good new is that in the way you suggested I got some more function than in the past.

I will try adding the "unpack hack" in redflat section, as I suppose the error I reported on prev comment was caused by a missing table.unpack in some file.

**Comment by worron** [comment]
>I just did a test on other confs too, but there I stil got some error reports

It's all right. Colored configs contain some settings for my current hardware and software environment. It's not designed to work with any random system, but more like example how you can improve colorless config.

You still can adapt colored configs to your PC, but some experience in Awesome WM configuration needed. Some things to pay attention to
* Variables in `env-config.lua`
* Widgets settings in "Panel widgets" section of main config. Do not forget to remove unwanted widgets from panel inside `awful.screen.connect_for_each_screen` function
* Desktop setup has a lot of widgets which are dependent on hardware, it's good idea to simply remove "Desktop widgets" for quick start
* For sure you want to change some personal settings in `autostart-config.lua`, `rules-config.lua`, `keys-config.lua`.

Also fonts, colors and general widget appearance can be changed in `~/.config/awesome/themes/color/theme.lua` files.

**Comment by ema79** [comment]
Yeah I noticed some setting as specific, the keyboard for example. As I noticed I had different error report for different machines. I'll keep in mind you advices btw.
Now I got colorless working should be easier to work on things to edit.

Many tanks

**Comment by ema79** [comment]
Hello Worron, during last days I had very few time to work on conf files ... consider I'm also very slow.
Must say your advices were very precious btw, they made me understand better how the config files are structurated.
For now just did a paste and copy of rc-coloress.lua and gave it a different name. Created the related folder, and loaded on my-own rc-coloress.lua desktop widgets for red themes. I've been able to make internet widget run changing the related hardware configuration ...
Next days I'll try some other improvement.

**Comment by worron** [comment]
Good luck. If your question solved the current issue should be closed i think. You always can reopen it or create a new one if another problem appears.
